app/user.go

```diff
@@ -2363,12 +2363,18 @@ func (a *App) UpdateThreadsReadForUser(userID, teamID string) *model.AppError {
 
 func (a *App) UpdateThreadFollowForUser(userID, teamID, threadID string, state bool) *model.AppError {
 	_, err := a.Srv().Store.Thread().MaintainMembership(userID, threadID, state, false, true, false, false)
+	if err != nil {
+		return model.NewAppError("UpdateThreadFollowForUser", "app.user.update_thread_follow_for_user.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	replyPosts, err := a.Srv().Store.Post().GetRepliesForExport(threadID)
```

Comment:
We should use `a.Srv().Store.Thread().Get(threadId)`  instead. Its a much faster store method than `GetRepliesForExport`

This line would change to:
`thread, err :=  a.Srv().Store.Thread().Get(threadId)`

and then use `thread.ReplyCount` below.